sentry-ruby/lib/sentry/hub.rb

```diff
@@ -144,6 +144,18 @@ def add_breadcrumb(breadcrumb, hint: {})
       current_scope.add_breadcrumb(breadcrumb)
     end
 
+    def generate_sentry_trace(span = nil)
+      return unless current_client.configuration.propagate_traces
+
+      span ||= current_scope.get_span
```

Comment:
@rhcarvalho let's move the discussion here.
Another reason I choose `Hub` instead of `Client` is because in `Hub` we can also access the current scope.

Consider that this method needs to access different both `Client` and `Scope`, I still think `Hub` is currently the best place to have this method.

Reply:
> https://github.com/getsentry/sentry-ruby/pull/1446#discussion_r636350633
> I'd say we don't need the Hub because it is only used to get the current scope when the span is nil, but we never call it with a nil span.

I have the impression this line is unnecessary, isn't it? The only non-test place where this method is called is [`set_sentry_trace_header`](https://github.com/getsentry/sentry-ruby/pull/1446/files#diff-e0361d5ec7d463e13a19fa414ba13a87d14ea9137f04a3e67515c4d854a5efaeR58-R61) and we never call with a `nil` span.

So my thought was that this didn't necessarily need to access the current scope.

Reply:
I thought being able to read the span from current scope is a hard requirement when I checked sentry-python's implementation. 

https://github.com/getsentry/sentry-python/blob/4c09f3203d6d19789c6fa729a2e46557ad4ea913/sentry_sdk/hub.py#L692-L694

But if it's not, I'm happy to remove it and move this method to `Client`.